src/Microsoft.PowerShell.PSReadLine/ReadLine.cs

```diff
@@ -888,33 +888,24 @@ public static void InvokePrompt(ConsoleKeyInfo? key = null, object arg = null)
         /// </summary>
         public static string GetPrompt()
         {
-            var runspaceIsRemote = _singleton._mockableMethods.RunspaceIsRemote(_singleton._runspace);
-            System.Management.Automation.PowerShell ps;
-            if (!runspaceIsRemote)
+            var command = new PSCommand();
+            command.AddCommand("prompt");
+            var results = HostUtilities.InvokeOnRunspace(command, _singleton._runspace);
```

Comment:
We'll be moving PSReadline to lzybkr/PSReadline - and when we do, the code needs to build against PowerShell V3, so this api won't be available.